src/oci/image/mod.rs

```diff
@@ -0,0 +1,98 @@
+pub mod config;
+pub mod layer;
+pub mod manifest;
+
+use std::error;
+
+pub trait Reference {
+    fn registry(self) -> Result<String, Box<dyn error::Error>>;
+
+    fn fullname(self) -> Result<String, Box<dyn error::Error>>;
+
+    fn tag(self) -> Result<String, Box<dyn error::Error>>;
+
+    fn digest(self) -> Result<String, Box<dyn error::Error>>;
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ImageReference(String);
+
+impl Reference for ImageReference {
+    fn registry(self) -> Result<String, Box<dyn error::Error>> {
```

Comment:
I can't see any error. 
Shouldn't it be an Option instead though?

Reply:
You're right, it's useless.